tests/test_indexes.py

```diff
@@ -113,6 +113,9 @@ def test_str_split() -> None:
     ind = pd.Index(["a-b", "c-d"])
     check(assert_type(ind.str.split("-"), "pd.Index[str]"), pd.Index)
     check(assert_type(ind.str.split("-", expand=True), pd.MultiIndex), pd.MultiIndex)
+    check(
+        assert_type(ind.str.split("-", expand=False), "pd.Index[list[str]]"), pd.Index
```

Comment:
```suggestion
        assert_type(ind.str.split("-", expand=False), "pd.Index[list[str]]"), pd.Index, list
```
This will make sure that what is in the `Index` are `list` types at runtime